packages/opencode/src/cli/cmd/tui.ts

```diff
@@ -104,11 +130,104 @@ export const TuiCommand = cmd({
         if (Object.keys(providers).length === 0) {
           return "needs_provider"
         }
+        const server = await (async () => {
+          if (!args.docker) {
+            return Server.listen({ port: args.port, hostname: args.hostname })
+          }
 
-        const server = Server.listen({
-          port: args.port,
-          hostname: args.hostname,
-        })
+          const docker = Bun.which("docker")
+          if (!docker) {
+            UI.error("docker not found, starting server locally")
+            return Server.listen({ port: args.port, hostname: args.hostname })
+          }
```

Comment:
For discussion if we want to allow this fallback. If I am depending on opencode to be isolated, I might miss that it failed and not notice if it just quietly falls back